62.CAD/vertex_shader.hlsl

```diff
@@ -398,20 +445,71 @@ PSInput main(uint vertexID : SV_VertexID)
-        otherwise it's just a second draw to "Resolve" and the only important thing on "Resolves" is the same `outV.position` as the previous draw (basically the same cage)
-        So if you do any precomputation, etc for sdf caluclations you could skip that :D and save yourself the trouble if `writeToAlpha` is false.
-    */
-    
-    
-// Make the cage fullscreen for testing:
+    else if (objType == ObjectType::CURVE_BOX)
+    {
+        outV.setColor(lineStyle.color);
+        outV.setLineThickness(screenSpaceLineWidth / 2.0f);
+
+        CurveBox curveBox;
+        curveBox.aabbMin = vk::RawBufferLoad<double2>(drawObj.geometryAddress, 8u);
+        curveBox.aabbMax = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2), 8u);
+        curveBox.curveMin[0] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 2, 8u);
+        curveBox.curveMin[1] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 3, 8u);
+        curveBox.curveMin[2] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 4, 8u);
+        curveBox.curveMax[0] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 5, 8u);
+        curveBox.curveMax[1] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 6, 8u);
+        curveBox.curveMax[2] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 7, 8u);
+
+        const bool2 maxCorner = bool2(vertexIdx & 0x1u, vertexIdx >> 1);
+        const double2 coord = transformPointNdc(lerp(curveBox.aabbMin, curveBox.aabbMax, maxCorner));
+        outV.position = float4((float2) coord, 0.f, 1.f);
+
+        const uint major = 1;
+        const uint minor = 1-major;
+
+        nbl::hlsl::shapes::QuadraticBezier curveMin = nbl::hlsl::shapes::QuadraticBezier::construct(
+            curveBox.curveMin[0], curveBox.curveMin[1], curveBox.curveMin[2]);
+        nbl::hlsl::shapes::QuadraticBezier curveMax = nbl::hlsl::shapes::QuadraticBezier::construct(
+            curveBox.curveMax[0], curveBox.curveMax[1], curveBox.curveMax[2]);
+
+        // Swizzled X = major
+        // Swizzled Y = minor
+        outV.uv = (float2) (major == 1 ? maxCorner.yx : maxCorner.xy);
+        outV.setCurveMinA(major == 1 ? curveMin.A().yx : curveMin.A().xy);
+        outV.setCurveMinB(major == 1 ? curveMin.B().yx : curveMin.B().xy);
+        outV.setCurveMinC(major == 1 ? curveMin.C().yx : curveMin.C().xy);
+        outV.setCurveMaxA(major == 1 ? curveMax.A().yx : curveMax.A().xy);
+        outV.setCurveMaxB(major == 1 ? curveMax.B().yx : curveMax.B().xy);
+        outV.setCurveMaxC(major == 1 ? curveMax.C().yx : curveMax.C().xy);
+
+        {
+            float a = outV.getCurveMinA().x;
+            float b = outV.getCurveMinB().x;
+            float c = outV.getCurveMinC().x - outV.uv.x;
+
+            float det = b*b-4.f*a*c;;
+            float rcp = 0.5f/a;
+            outV.minCurveDetRcp2_BRcp = float2(
```

Comment:
do outV.setXXX makes it easier to change the layout of stuff internally later on